src/dbnode/storage/series/types.go

```diff
@@ -95,24 +105,26 @@ type DatabaseSeries interface {
 	// Bootstrap merges the raw series bootstrapped along with any buffered data.
 	Bootstrap(blocks block.DatabaseSeriesBlocks) (BootstrapResult, error)
 
-	// Flush flushes the data blocks of this series for a given start time
-	Flush(
+	// WarmFlush flushes the WarmWrites of this series for a given start time.
+	WarmFlush(
 		ctx context.Context,
 		blockStart time.Time,
 		persistFn persist.DataFn,
-		version int,
 		nsCtx namespace.Context,
 	) (FlushOutcome, error)
 
 	// Snapshot snapshots the buffer buckets of this series for any data that has
-	// not been rotated into a block yet
+	// not been rotated into a block yet.
 	Snapshot(
 		ctx context.Context,
 		blockStart time.Time,
 		persistFn persist.DataFn,
 		nsCtx namespace.Context,
 	) error
 
+	// NeedsColdFlushBlockStarts returns the block starts that need cold flushes.
+	NeedsColdFlushBlockStarts() OptimizedTimes
```

Comment:
Other methods we have called `Needs...` return a bool (e.g. `NeedsFlush(time) bool`), would be better to call this something like `ColdFlushBlockStarts`?

Reply:
Yes that definitely makes more sense 👍